### PR TITLE
Make contract deployment more atomic and less frontrunnable

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Adds constructor args to `MultisigIsm`

### Testing

Unit Tests

### CODE REVIEW

File: solidity/contracts/isms/MultisigIsm.sol [hunk 1]
```diff
@@ -82,33 +82,26 @@ contract MultisigIsm is IMultisigIsm, Ownable {
...
-    // ============ External Functions ============
-
-    /**
-     * @notice Enrolls multiple validators into a validator set.
-     * @dev Reverts if `_validator` is already in the validator set.
-     * @param _domains The remote domains of the validator sets.
-     * @param _validators The validators to add to the validator sets.
-     * @dev _validators[i] are the validators to enroll for _domains[i].
-     */
-    function enrollValidators(
```
Comment (nambrot) [inline solidity/contracts/isms/MultisigIsm.sol:98]: Wouldn't this removal mean that you can't enroll validators afterwords by the owner?

**Comment by asaj** [comment]
What's the motivation behind this?

File: solidity/contracts/isms/MultisigIsm.sol (hunk 1 above)
Comment (yorhodes) [reply solidity/contracts/isms/MultisigIsm.sol:98]: happy to leave these
basically didn't like the API of nested arrays

Comment (asaj) [reply solidity/contracts/isms/MultisigIsm.sol:98]: Man you really hate nested arrays don't you

Comment (asaj) [reply solidity/contracts/isms/MultisigIsm.sol:98]: I don't mind the struct change but making the MultisigIsm validator configs immutable post-initialization is something that merits more discussion

Comment (yorhodes) [reply solidity/contracts/isms/MultisigIsm.sol:98]: they're not immutable, I just got rid of the batch functions

**Comment by nambrot** [comment]
@yorhodes what's the state of this PR?

**Pull request closed without merging**